pkg/shed/db.go

```diff
@@ -74,83 +87,372 @@ func NewDB(path string, logger logging.Logger) (db *DB, err error) {
+// Put inserts the given key and value in to badger.
 func (db *DB) Put(key []byte, value []byte) (err error) {
-	err = db.ldb.Put(key, value, nil)
-	if err != nil {
-		db.logger.Debugf("failed to insert in to DB. Error : %s",  err.Error())
-		db.metrics.PutFailCounter.Inc()
-		return err
-	}
-	db.metrics.PutCounter.Inc()
-	return nil
+	db.logger.Tracef("Put value len : %d for key %s", len(value), string(key))
+	return db.bdb.Update(func(txn *badger.Txn) (err error) {
+		db.metrics.PutCount.Inc()
+		err = txn.Set(key, value)
+		if err != nil {
+			db.logger.Errorf("Could not insert chunk with key %s in DB", string(key))
+			db.metrics.PutFailCount.Inc()
+			return err
+		}
+		db.logger.Tracef(" put success with key %s and value len %d", string(key), len(value))
+		return nil
+	})
+
 }
 
-// Get wraps LevelDB Get method to increment metrics counter.
+// Get retrieves the value given the key.
+// if the key is not present a ErrNotFound is returned.
 func (db *DB) Get(key []byte) (value []byte, err error) {
-	value, err = db.ldb.Get(key, nil)
-	if err == leveldb.ErrNotFound {
-		db.logger.Debugf("key %s not found during GET", string(key))
-		db.metrics.GetNotFoundCounter.Inc()
-		return nil, err
-	} else {
+	db.logger.Tracef("Get value for key %s", string(key))
+	err = db.bdb.View(func(txn *badger.Txn) (err error) {
+		item, err := txn.Get(key)
+		if err != nil {
+			db.metrics.GetFailCount.Inc()
+			if err == badger.ErrKeyNotFound {
+				db.metrics.GetNotFoundCount.Inc()
+				return ErrNotFound
+			}
+			db.metrics.GetFailCount.Inc()
+			return err
+		}
+		return item.Value(func(val []byte) error {
+			value = val
+			db.logger.Tracef("got valeu for key %s with length %d", string(key), len(val))
+			return nil
+		})
+	})
+	if err != nil {
 		db.logger.Errorf("key %s not found in DB", string(key))
-		db.metrics.GetFailCounter.Inc()
+	} else {
+		db.logger.Tracef("got value with len : %d for key : %s", len(value), string(key))
+		db.metrics.GetCount.Inc()
 	}
-	db.metrics.GetCounter.Inc()
-	return value, nil
+	return value, err
 }
 
-// Has wraps LevelDB Has method to increment metrics counter.
+// Has checks if the given key is present in the database.
+// it returns a bool indicating true or false OR error if it encounters one during the operation.
 func (db *DB) Has(key []byte) (yes bool, err error) {
-	yes, err = db.ldb.Has(key, nil)
+	db.logger.Tracef("Has value with key %s", string(key))
+	yes = false
+	err = db.bdb.View(func(txn *badger.Txn) (err error) {
+		item, err := txn.Get(key)
+		if err != nil {
+			if err == badger.ErrKeyNotFound {
+				return ErrNotFound
+			}
+			return err
+		}
+		return item.Value(func(val []byte) error {
+			yes = true
+			db.metrics.HasCount.Inc()
+			return nil
+		})
+	})
 	if err != nil {
-		db.logger.Debugf("encountered error during HAS of key %s. Error: %s ", string(key), err.Error())
-		db.metrics.HasFailCounter.Inc()
-		return false, err
+		db.logger.Errorf("key %s not found in DB", string(key))
```

Comment:
This log line is not needed.

The error is not returned at all please return it, and remove the log line.